bridge/src/reactor.rs

```diff
@@ -0,0 +1,124 @@
+use crate::connection::{Connection, ConnectionError};
+use tokio::runtime::Builder;
+use tokio::sync::{mpsc, oneshot};
+
+#[derive(Debug)]
+pub enum Request {
+    Connect {
+        host: String
+    },
+    Rpc {
+        connection: Connection,
+        method: String,
+        bytes: Vec<u8>
+    }
+}
+
+#[derive(Debug)]
+pub enum Response {
+    Empty {},
+    Connection {
+        connection: Connection
+    },
+    Rpc {
+        bytes: Vec<u8>
+    }
+}
+
+#[derive(Debug)]
+pub enum ReactorError {
+    ConnectionError(ConnectionError)
+}
+
+impl std::fmt::Display for ReactorError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ReactorError::ConnectionError(error) => write!(f, "{}", error),
+        }
+    }
+}
+
+impl From<ConnectionError> for ReactorError {
+    fn from(err: ConnectionError) -> Self {
+        ReactorError::ConnectionError(err)
+    }
+}
+
+impl std::error::Error for ReactorError {}
+
+pub type ReactorResult = Result<Response, ReactorError>;
+
+// A request wrapper which includes a channel for sending back the response
+#[derive(Debug)]
+struct Envelope {
+    request: Request,
+    channel: oneshot::Sender<ReactorResult>
+}
+
+async fn handle_request(request: Request, channel: oneshot::Sender<ReactorResult>) {
```

Comment:
The args to this fn could just be an `Envelope`

Reply:
What do you think it achieves? I'm also assuming it's a nit